### PR TITLE
Remove client-service Asio communication

### PR DESCRIPTION
**Pull request by dpallotti**
#### Remove FLAGS_asio_port on the client and the code that used it

---
@antonrohr @beckerhe Even if in a minor way, I touched code related to ssh and to the watchdog, so please confirm that things still look in order.

### CODE REVIEW

File: OrbitQt/main.cpp [hunk 1]
```diff
@@ -328,7 +305,7 @@ int main(int argc, char* argv[]) {
 
     while (true) {
       const auto result = RunUiInstance(&app, deployment_configuration,
-                                        &(context.value()), options);
+                                        &(context.value()), ApplicationOptions{});
```
Comment (dimitry-) [inline OrbitQt/main.cpp:308]: maybe just remove the parameter?

Comment (dpallotti) [reply OrbitQt/main.cpp]: Yes, done.

File: OrbitQt/orbitwatchwidget.cpp [hunk 2]
```diff
@@ -275,7 +275,7 @@ void OrbitWatchWidget::OnUpdateWatch(const Variable* a_Variable) {
 void OrbitWatchWidget::on_FindLineEdit_textChanged(const QString& /*arg1*/) {}
 
 //-----------------------------------------------------------------------------
-void OrbitWatchWidget::on_RefreshButton_clicked() { GOrbitApp->RefreshWatch(); }
+void OrbitWatchWidget::on_RefreshButton_clicked() {}
```
Comment (beckerhe) [inline OrbitQt/orbitwatchwidget.cpp:278]: It should be fine to remove this function entirely.

File: OrbitQt/servicedeploymanager.h [hunk 3]
```diff
@@ -26,7 +26,6 @@ class ServiceDeployManager : public QObject {
 
  public:
   struct Ports {
```
Comment (beckerhe) [inline OrbitQt/servicedeploymanager.h:28]: Maybe we should rename that struct to "GrpcPort" and use it as a strong typedef.

File: OrbitQt/main.cpp [hunk 4]
```diff
@@ -140,10 +123,8 @@ static outcome::result<void> RunUiInstance(
   }());
   const auto& [ports, capture_path] = result;
 
-  options.asio_server_address =
-      absl::StrFormat("127.0.0.1:%d", ports.asio_port);
-  options.grpc_server_address =
-      absl::StrFormat("127.0.0.1:%d", ports.grpc_port);
+  ApplicationOptions options{
+      .grpc_server_address = absl::StrFormat("127.0.0.1:%d", ports.grpc_port)};
```
Comment (beckerhe) [inline OrbitQt/main.cpp:127]: Unfortunately not yet supported on the VS toolset that we use on the CI.

Comment (dpallotti) [reply OrbitQt/main.cpp]: Ah right, we have other code that does this but it's Linux only.

File: OrbitQt/orbitwatchwidget.cpp (hunk 2 above)
Comment (dpallotti) [reply OrbitQt/orbitwatchwidget.cpp]: Yep, done.

File: OrbitQt/servicedeploymanager.h (hunk 3 above)
Comment (dpallotti) [reply OrbitQt/servicedeploymanager.h]: Agree, done.

**Pull request merged**